build/targets.mk

```diff
@@ -314,6 +314,12 @@ endif
 ifeq ($(TARGET),ANDROID)
   ANDROID_NDK ?= $(HOME)/opt/android-ndk-r20-beta3
 
+  # Control if the Java and other machine independent parts are being built
+  # or if the native libxcsoar.so is being built.
+  # The caller must never set this 'n'. This is being done by make itself
+  # which calls itself with 'ANDROID_BUILD_APK = n' to build the native lib.
+  ANDROID_BUILD_APK ?= y
```

Comment:
`make` has a built-in syntax to declare which targets shall be built. This sounds like it attempts to imitate this `make` feature, but badly?

Reply:
Removed the use this variable to (re) define the make targets. It is still used internally to distinguish between the outer and inner make.